File: mobile/src/main/java/com/rndapp/roostremote/models/StringOption.java
package com.rndapp.roostremote.models;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.List;

/**
 * Created by ell on 1/3/17.
 */

public class StringOption extends Option {
    protected String name;
    protected String value;

    public StringOption(String name, String value) {
        this.name = name;
        this.value = value;
    }

    @Override
    public JSONObject getJsonObject(String key) throws JSONException {
        JSONObject jsonObject = new JSONObject();
        jsonObject.put(key, value);
        return jsonObject;
    }

    @Override
    public String getName() {
        return name;
    }

    public String getValue() {
        return value;
    }
}

File: mobile/src/main/java/com/rndapp/roostremote/models/StringOption.kt
package com.rndapp.roostremote.models

import org.json.JSONException
import org.json.JSONObject

/**
 * Created by ell on 1/3/17.
 */

class StringOption(name: String, value: String) : Option(name, value) {

    @Throws(JSONException::class)
    override fun getJsonObject(key: String): JSONObject {
        val jsonObject = JSONObject()
        jsonObject.put(key, value as String)
        return jsonObject
    }
}
